sdk-core/src/main/java/dev/restate/sdk/core/GrpcContextDataProvider.java

```diff
@@ -24,17 +26,16 @@ public Map<String, String> supplyContextData() {
     InvocationId invocationId = InvocationId.INVOCATION_ID_KEY.get();
     String serviceMethod = RestateGrpcServer.SERVICE_METHOD.get();
 
-    // We use Map.of constructors to avoid allocating hashmaps
-    if (invocationId == null && serviceMethod == null) {
-      return Map.of();
-    } else if (invocationId == null) {
-      return Map.of(RestateGrpcServer.LoggingContextSetter.SERVICE_METHOD_KEY, serviceMethod);
-    } else {
-      return Map.of(
-          RestateGrpcServer.LoggingContextSetter.INVOCATION_ID_KEY,
-          invocationId.toString(),
-          RestateGrpcServer.LoggingContextSetter.SERVICE_METHOD_KEY,
-          serviceMethod);
+    var context = new HashMap<>(ThreadContext.getContext());
```

Comment:
I don't fully understand why we need to propagate the ThreadContext here. Will log4j2 just merge all the different context providers? https://logging.apache.org/log4j/2.x/manual/extending.html#custom-contextdataprovider

Can you also test this change works as expected with the http endpoint?

Reply:
I also read that page but it didn't answer the question of multiple providers co-existing.

In my testing, log4j did NOT merge the contexts – only the two keys above were available prior to my changes. Will re-test and confirm though.

Reply:
ok then let's go ahead with this but let's verify nothing breaks with http endpoints. After release let's re-evaluate this bit about logging context integration.

Reply:
Removed altogether as discussed - thanks for your help! It only breaks for handler-emitted log entries from Java, but not from Kotlin, as expected.